webapp/components/post_view/components/new_message_indicator.jsx

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+export default class NewMessageIndicator extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            visible: false,
+            rendered: false
+        };
+    }
+    componentDidMount() {
+        this.onTransition = this.setRendered.bind(this);
+        this.refs.indicator.addEventListener('transitionend', this.onTransition);
+    }
+    componentWillUnmount() {
+        this.refs.indicator.removeEventListener('transitionend', this.onTransition);
+    }
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.newMessages > 0) {
+            this.setState({rendered: true}, () => {
+                this.setState({visible: true});
+            });
+        } else {
+            this.setState({visible: false});
+        }
+    }
+    render() {
+        let className = 'nav-pills__unread-indicator-bottom';
+        if (this.state.visible > 0) {
```

Comment:
True, in the regular view postlistview, this isn't necessary.. 
While developing I encountered some views where the absolute positioning of the indicator influenced the scrollheight of the parent. I will double check if this is still the case, otherwise this logic might be a bit over-the-top indeed...

Then again, the display: none makes this component more fail-proof; hidden when it's irrelevant. :)